src/features/payment/components/IframeCardForm/IframeCardForm.tsx

```diff
@@ -0,0 +1,289 @@
+import React from "react";
+import { Box } from "@mui/material";
+import { useTranslation } from "react-i18next";
+import {
+  getSessionItem,
+  SessionItems,
+} from "../../../../utils/storage/sessionStorage";
+import { FormButtons } from "../../../../components/FormButtons/FormButtons";
+import { PaymentMethod } from "../../../../features/payment/models/paymentModel";
+import { Field, RenderField } from "./IframeCardField";
+interface Props {
+  loading?: boolean;
+  onCancel: () => void;
+  onSubmit?: (bin: string) => void;
+  hideCancel?: boolean;
+}
+
+interface FieldFormStatus {
+  isValid: boolean;
+  errorCode: null | string;
+  errorMessage: null | string;
+}
+
+interface BuildResp {
+  sessionId: string;
+  securityToken: string;
+  fields: Array<Field>;
+}
+
+export type IdFields =
+  | "CARD_NUMBER"
+  | "EXPIRATION_DATE"
+  | "SECURITY_CODE"
+  | "CARDHOLDER_NAME";
+
+type FieldsFormStatus = Map<IdFields, FieldFormStatus>;
+
+const initialFormStatus: FieldFormStatus = {
+  isValid: false,
+  errorCode: null,
+  errorMessage: null,
+};
+
+const fieldFormStatus: FieldsFormStatus = new Map();
+fieldFormStatus.set("CARD_NUMBER", initialFormStatus);
+fieldFormStatus.set("EXPIRATION_DATE", initialFormStatus);
+fieldFormStatus.set("SECURITY_CODE", initialFormStatus);
+fieldFormStatus.set("CARDHOLDER_NAME", initialFormStatus);
+
+// eslint-disable-next-line functional/no-let
+let formStatus = false;
+// eslint-disable-next-line functional/no-let
+let sdkBuildInstance: { confirmData: () => void };
+
+// eslint-disable-next-line sonarjs/cognitive-complexity
+export default function IframeCardForm(props: Props) {
+  const { loading = true, onCancel, onSubmit = () => null, hideCancel } = props;
+  const [error, setError] = React.useState(false);
+  const [form, setForm] = React.useState<BuildResp>();
+  const [spinner, setSpinner] = React.useState(loading);
+  // this dummy state is only used to perform a component update, not the best solution but works
+  const [, setDummyState] = React.useState(0);
+
+  const { hostname, protocol, port } = window.location;
+
+  const calculateFormValidStatus = (
+    fieldFormStatus: Map<string, FieldFormStatus>
+  ) =>
+    [
+      fieldFormStatus.get("CARD_NUMBER")?.isValid,
+      fieldFormStatus.get("EXPIRATION_DATE")?.isValid,
+      fieldFormStatus.get("SECURITY_CODE")?.isValid,
+      fieldFormStatus.get("CARDHOLDER_NAME")?.isValid,
+    ].every((isValid) => isValid);
+
+  React.useEffect(() => {
+    if (!form) {
```

Comment:
All this logic present in the React.useEffect method, can we move it to the **helper.ts** component to maintain consistency with the rest of the code? _(the **helper.ts** component contains all the logic to invoke the **eCommerce** services)_ 🚀
Also to invoke the service we can generate the client code. I created a PR on infra with a new branch useful for this purpose 👉🏻 https://github.com/pagopa/pagopa-infra/pull/1241  👀

_The branch contains only the code to generate the client of the `/ecommerce/checkout/v1/payment-methods/${paymentMethodId}/preauthorization` call_